src/Common/src/Interop/Windows/Winsock/Interop.WSARecv.cs

```diff
@@ -11,33 +11,23 @@ internal static partial class Interop
     internal static partial class Winsock
     {
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
-        internal static extern SocketError WSARecv(
-            [In] SafeCloseSocket socketHandle,
-            [In] ref WSABuffer buffer,
-            [In] int bufferCount,
-            [Out] out int bytesTransferred,
-            [In, Out] ref SocketFlags socketFlags,
-            [In] SafeHandle overlapped,
-            [In] IntPtr completionRoutine);
-
-        [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
-        internal static extern SocketError WSARecv(
-            [In] SafeCloseSocket socketHandle,
-            [In, Out] WSABuffer[] buffers,
-            [In] int bufferCount,
-            [Out] out int bytesTransferred,
-            [In, Out] ref SocketFlags socketFlags,
-            [In] SafeHandle overlapped,
-            [In] IntPtr completionRoutine);
+        internal static extern unsafe int WSARecv(
+            SafeCloseSocket socketHandle,
+            WSABuffer* buffers,
+            int bufferCount,
+            int* bytesTransferred,
+            SocketFlags* socketFlags,
+            SafeNativeOverlapped overlapped,
+            void* completionRoutine);
```

Comment:
As an aside, our DllImport guidance we've tried to follow is that the parameter names should match the names used on the native side, e.g. this would end up being:
```C#
internal static extern unsafe int WSARecv(
    SafeCloseSocket s,
    WSABuffer* lpBuffers,
    int dwBufferCount,
    int* lpNumberOfBytesRecvd,
    SocketFlags* lpFlags,
    SafeNativeOverlapped lpOverlapped,
    void* lpCompletionRoutine);
```
almost as if we didn't have to declare the P/Invokes manually and they were just automatically generated from a header file.  We've tried to follow that guidance for new P/Invokes we write, but in some mass import cases from desktop, they were never cleaned up.  As long as you're touching them here, might be worthwhile fixing them up accordingly.